### PR TITLE
No contacts in component list command result.

### PR DESCRIPTION
**Pull request by xychu**
JIRA: PDC-1166

### CODE REVIEW

File: tests/component/data/global_component/list.json [hunk 1]
```diff
@@ -1,38 +1 @@
...
-                "contact_role": "test_role_b",
-                "contact": {
-                    "username": "Test User",
-                    "email": "test_user@example.com"
-                }
-            }
-        ]
-    }
-]
+[{"name": "Test Global Component", "contacts": [], "labels": [{"url": "http://example.com/lable/1", "name": "test label", "description": "test label description"}], "dist_git_web_url": "http://pkgs.example.com/test_global_component", "upstream": {"homepage": "http://test_global_component.org", "scm_url": "http://test_global_component.org/git", "scm_type": "git"}, "dist_git_path": null, "id": "1"}]
```
Comment (simozhan) [inline tests/component/data/global_component/list.json]: it is better to format it

**Comment by simozhan** [comment]
+1. Looks good to me

**Comment by lubomir** [comment]
For a list operation, the contacts are only used in the JSON output, right? If so, they should be only downloaded if required. However, maybe it would make sense not to include them at all.

File: pdc_client/plugins/component.py [hunk 2]
```diff
@@ -64,8 +64,15 @@ def list_global_components(self, args):
...
+            component_contacts = get_paged(self.client['global-component-contacts']._,
+                                           component=global_component['name'])
+            update_component_contacts(global_component, component_contacts)
+
         if args.json:
-            print json.dumps(list(global_components))
+            print json.dumps(global_components)
             return
 
         if global_components:
```
Comment (ycheng-aa) [inline pdc_client/plugins/component.py:72]: If not args.json, the contact information won't be used at all, but above already got them. If global_components already is a list, 'if global_components" looks unnecessary.

File: pdc_client/plugins/component.py [hunk 3]
```diff
@@ -181,8 +188,16 @@ def list_release_components(self, args):
...
+                                           component=release_component['name'],
+                                           release=self._get_release_id(release_component))
+            update_component_contacts(release_component, component_contacts)
+
         if args.json:
-            print json.dumps(list(release_components))
+            print json.dumps(release_components)
             return
 
         if release_components:
```
Comment (ycheng-aa) [inline pdc_client/plugins/component.py:201]: Same as for global component.

**Comment by xychu** [comment]
I vote for not including the contacts by default unless they are required.
And just realize that with the replacement, the old contact related filters('contact-role' and 'email') in list op will make not that much sense.

**Comment by xychu** [comment]
After we remove the old contacts, now there's no contacts info in component list op result.

**Comment by ycheng-aa** [comment]
Looks good.

**Pull request merged**